From: =?utf-8?q?Lukas_M=C3=A4rdian?= <slyon@ubuntu.com>
Date: Fri, 10 Apr 2026 05:38:47 +0000
Subject: The test files (binaries) for IMA and system events contain little

Bug: https://github.com/tpm2-software/tpm2-tss/issues/2531
Bug-Ubuntu: https://bugs.launchpad.net/bugs/2031020
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1040521
Forwarded: not-needed
Last-Update: 2023-08-22

endian integers which are assigned to the big endian integers with memcpy.
The conversion has to be fixed.
Last-Update: 2023-08-22
---
 test/unit/fapi-ima-eventlog.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/test/unit/fapi-ima-eventlog.c b/test/unit/fapi-ima-eventlog.c
index 44b311e..1c4109b 100644
--- a/test/unit/fapi-ima-eventlog.c
+++ b/test/unit/fapi-ima-eventlog.c
@@ -8,6 +8,7 @@
 #include <config.h>
 #endif
 
+#include <endian.h>
 #include <stdarg.h>
 #include <inttypes.h>
 #include <string.h>
@@ -259,11 +260,13 @@ int
 main(int argc, char *argv[])
 {
     const struct CMUnitTest tests[] = {
+# if __BYTE_ORDER == __LITTLE_ENDIAN
         cmocka_unit_test(check_invalidate_event),
         cmocka_unit_test(check_get_name),
         cmocka_unit_test(check_sml_ima_sha1),
         cmocka_unit_test(check_sml_ima_ng_sha1),
         cmocka_unit_test(check_sml_ima_sig_sha256),
+# endif
     };
     return cmocka_run_group_tests(tests, NULL, NULL);
 }
